src/main/kotlin/com/varabyte/kobweb/intellij/colors/KobwebColorProvider.kt

```diff
@@ -0,0 +1,222 @@
+                ?: return@apply
+
+            when {
+                callee.isKobwebColorFunction("rgb(r: Int, g: Int, b: Int)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Int>(3) ?: return@run
+                    return safeRgbColor(args[0], args[1], args[2])
+                }
+
+                callee.isKobwebColorFunction("rgb(value: Int)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Int>(1) ?: return@run
+                    return safeRgbColor(args[0])
+                }
+
+                callee.isKobwebColorFunction("rgba(value: Int)", "rgba(value: Long)") -> run {
+                    val args = element.valueArguments.run {
+                        evaluateArguments<Int>(1) ?: evaluateArguments<Long, Int>(1) { it.toInt() }
+                    } ?: return@run
+                    return safeRgbColor(args[0] shr Byte.SIZE_BITS)
+                }
+
+                callee.isKobwebColorFunction("argb(value: Int)", "argb(value: Long)") -> run {
+                    val args = element.valueArguments.run {
+                        evaluateArguments<Int>(1) ?: evaluateArguments<Long, Int>(1) { it.toInt() }
+                    } ?: return@run
+                    return safeRgbColor(args[0] and 0x00_FF_FF_FF)
+                }
+
+                callee.isKobwebColorFunction("hsl(h: Float, s: Float, l: Float)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Float>(3) ?: return@run
+                    return safeHslColor(args[0], args[1], args[2])
+                }
+
+                callee.isKobwebColorFunction("hsla(h: Float, s: Float, l: Float, a: Float)") -> run {
+                    val args = element.valueArguments.evaluateArguments<Float>(4) ?: return@run
+                    return safeHslColor(args[0], args[1], args[2])
+                }
+            }
+        }
+
+        else -> null
+    }
+
+    return if (currentDepth <= MAX_SEARCH_DEPTH) {
+        nextElement?.let { traceColor(it, currentDepth + 1) }
+    } else null
+}
+
+/**
+ * Pretty much emulates what CTRL-clicking on a reference would do
+ */
+private fun KtSimpleNameExpression.locateSource(): PsiElement? {
+    val resolvedMainReference = this.mainReference.resolve() ?: return null
+    val sourceDescriptor =
+        PsiNavigationSupport.getInstance().getDescriptor(resolvedMainReference) as? OpenFileDescriptor ?: return null
+    val sourcePsiFile = sourceDescriptor.file.toPsiFile(this.project) ?: return null
+    val sourceElementIdentifier = PsiUtilBase.getElementAtOffset(sourcePsiFile, sourceDescriptor.offset)
+    val sourceElement = sourceElementIdentifier.parent
+
+    return sourceElement
+}
+
+/**
+ * Uses the kotlin analysis api, as it can parse the constants much smarter.
+ * It can parse decimal, hex and binary automatically for example.
```

Comment:
It might be really nice if you could give a real, concrete example. Especially useful with all these abstract Kotlin classes flying around.

Maybe like "For example, the code token `0xF3`, if one of the arguments in the list, would get converted to its raw hex value in the output"